src/UniOneClient.php

```diff
@@ -11,110 +11,134 @@
+        $this->httpClient = new Client([
+          'base_uri' => $this::$endpoint,
+        ]);
+    }
 
-  /**
-   * Set the API key.
-   *
-   * @param string $apiKey The API key.
-   */
-  public function setApiKey(string $apiKey): void {
-    $this->apiKey = $apiKey;
-  }
+    /**
+     * Set the API key.
+     *
+     * @param string $apiKey the API key
+     */
+    public function setApiKey(string $apiKey): void
+    {
+        $this->apiKey = $apiKey;
+    }
 
-  /**
-   * Send a request to the UniOne API.
-   *
-   * @param array $body The request body.
-   * @param array $head The request headers.
-   *
-   * @return string
-   * @throws \GuzzleHttp\Exception\GuzzleException
-   */
-  public function sender(array $body, array $head = []): string {
-    $responseHeader = [
-      'Content-Type' => 'application/json',
-      'Accept' => 'application/json',
-      'X-API-KEY' => $this->apiKey,
-      'X-Mailer' => 'UniOne SDK-PHP agent'
-    ];
+    /**
+     * Set the timeout in seconds.
+     *
+     * @param float $timeout the timeout in seconds
+     */
+    public function setTimeout(float $timeout): void
+    {
+        $this->timeout = $timeout;
+    }
 
-    // Build body for request.
-    // @todo: Refactor this and create a special class for it.
-    $responseBody = [
-      "message" => [
-        "recipients" => [
-          [
-            "email" => $body['recipients_email'] ?? NULL,
-            "substitutions" => $body['recipients_substitutions'] ?? NULL,
-            "metadata" => $body['recipients_metadata'] ?? NULL,
-          ]
-        ],
-        "template_id" => $body['template_id'] ?? NULL,
-        "skip_unsubscribe" => $body['skip_unsubscribe'] ?? NULL,
-        "global_language" => $body['global_language'] ?? NULL,
-        "template_engine" => $body['template_engine'] ?? NULL,
-        "global_substitutions" => $body['global_substitutions'] ?? NULL,
-        "global_metadata" => $body['global_metadata'] ?? NULL,
-        "body" => [
-          "html" => $body['body_html'] ?? NULL,
-          "plaintext" => $body['body_plaintext'] ?? NULL,
-          "amp" => $body['body_amp'] ?? NULL,
-        ],
-        "subject" => $body['subject'] ?? NULL,
-        "from_email" => $body['from_email'] ?? NULL,
-        "from_name" => $body['from_name'] ?? NULL,
-        "reply_to" => $body['reply_to'] ?? NULL,
-        "track_links" => 0,
-        "track_read" => 0,
-        "headers" => $body['headers'] ?? NULL,
-        "attachments" => $body['attachments'] ?? NULL,
-        "inline_attachments" => $body['inline_attachments'] ?? NULL,
-        "options" => $body['options'] ?? NULL,
-      ]
-    ];
+    /**
+     * Send a request to the UniOne API.
+     *
+     * @param array $body the request body
+     * @param array $head the request headers
+     *
+     * @return string
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function sender(array $body, array $head = []): string
```

Comment:
It's more common to use the word `header` instead of `head`, so I would rename 2nd argument to `$headers`.